org.eclipse.dawnsci.analysis.dataset.test/src/org/eclipse/dawnsci/analysis/dataset/FunctionalUtilsTest.java

```diff
@@ -208,4 +217,36 @@ public void testDatasetDoubleStream() {
 			testDatasetDoubleStream(v, true);
 		}
 	}
+	
+	@Test
+	public void testLazyDatasetStream() {
+		Dataset original = DatasetFactory.createRange(DoubleDataset.class,10);
+		Dataset a = original.getSlice().reshape(new int[]{1,1,10});
+		a = DatasetUtils.tile(a, new int[]{4,5,1});
+		Stream<ILazyDataset> lzStream = FunctionalUtils.createLazyDatasetStream(a, new int[]{0, 1}, false);
+		doTest(lzStream.map(d -> safeSlice(d)), original);
+	}
+	
+	@Test
+	public void testDatasetStream() {
+		Dataset original = DatasetFactory.createRange(DoubleDataset.class,10);
+		Dataset a = original.getSlice().reshape(new int[]{1,1,10});
+		a = DatasetUtils.tile(a, new int[]{4,5,1});
+		Stream<IDataset> lzStream = FunctionalUtils.createDatasetStream(a, new int[]{0, 1}, false);
+		doTest(lzStream,original);
+	}
+	
+	private void doTest(Stream<IDataset> stream, Dataset original) {
+		List<Double> collect = stream.map(d -> d.max().doubleValue()).collect(Collectors.toList());
+		Dataset d = DatasetFactory.createFromList(collect);
+		Assert.assertTrue(Comparisons.equalTo(original, d).all());
+	}
+	
+	private static IDataset safeSlice(ILazyDataset lz) {
+		try {
+			return lz.getSlice();
+		} catch (DatasetException e) {
+			throw new RuntimeException("Couldn't slice data");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)